shell/platform/android/image_external_texture_gl.cc

```diff
@@ -29,19 +29,19 @@ ImageExternalTextureGL::ImageExternalTextureGL(
 
 void ImageExternalTextureGL::Attach(PaintContext& context) {
   if (state_ == AttachmentState::kUninitialized) {
-    if (!latest_android_image_.is_null() && !latest_bounds_.isEmpty()) {
-      // After detach the cache of textures will have been cleared. If
-      // there is an android image we must populate it now so that the
-      // first frame isn't blank.
-      JavaLocalRef hardware_buffer = HardwareBufferFor(latest_android_image_);
-      UpdateImage(hardware_buffer, context);
-      CloseHardwareBuffer(hardware_buffer);
-    }
+    // TODO(johnmccurtchan): We currently display the first frame after an
+    // attach-detach cycle as blank. There seems to be an issue on some
+    // devices where ImageReaders/Images from before the detach aren't
+    // valid after the attach. According to Android folks this doesn't
+    // match the spec. Revisit this in the future.
+    // See https://github.com/flutter/flutter/issues/142978 and
+    // https://github.com/flutter/flutter/issues/139039.
```

Comment:
We could probably save a rasterized version somewhere to display when we wake up, but it wouldn't be very fast and it would use a bunch of memory.

Reply:
Seems like the spec should allow us to cache the Image as we were so I'd rather investigate that than go down a path of preserving the texture contents in the C heap.

Reply:
We'd also need to handle if the image is a protect format, as IIRC those cannot be readback.